app/models/fixed_asset.rb

```diff
@@ -375,26 +376,30 @@ def self.depreciate(options = {})
 
   def depreciate!
     planned_depreciations.clear
-    # Computes periods
-    starts = [started_on, self.stopped_on + 1]
-    starts += depreciations.pluck(:started_on)
-
-    first_day_of_month = ->(date) { date.day == 1 } # date.succ.day < date.day }
-    new_months = (started_on...stopped_on).select(&first_day_of_month)
-
-    case depreciation_period
-    when /monthly/
-      starts += new_months
-    when /quarterly/
-      new_trimesters = new_months.select { |date| date.month.multiple_of? 3 }
-      starts += new_trimesters
-    when /yearly/
-      new_years = new_months.select { |date| date.month == 1 }
-      starts += new_years
+
+    unless depreciation_method.to_s == 'none'
```

Comment:
Can we add `predicates: { prefix: true }` to the enumerize and then replace this with `depreciation_method_none?` ?